pkg/api/types.go

```diff
@@ -2379,6 +2380,12 @@ type DeleteOptions struct {
 	// Should the dependent objects be orphaned. If true/false, the "orphan"
 	// finalizer will be added to/removed from the object's finalizers list.
 	OrphanDependents *bool `json:"orphanDependents,omitempty"`
+
+	// Should the owner object keep existing in the key-value store until all dependents are removed.
+	// The option is cascading, i.e., the object’s dependents will be deleted with the same SynchronousGarbageCollection option.
+	// This option and OrphanDependents are exclusive.
+	// Defaults to false.
+	SynchronousGarbageCollection *bool
```

Comment:
I thought the graceful deletion issue still hadn't be resolved.  I had mentioned the idea of a timeout where the user gets to choose how long to wait.  That would make this a `*int`.

I remember you had a comment back about impersonation not being effective, so the issue is still outstanding, correct?